#!/bin/sh /usr/share/dpatch/dpatch-run
## 03-pidfile.dpatch

@DPATCH@

diff -Naur pxe-1.4.2.BAR/pxe.cc pxe-1.4.2.FOO/pxe.cc
--- pxe-1.4.2.BAR/pxe.cc	2006-08-15 10:12:25.000000000 +0000
+++ pxe-1.4.2.FOO/pxe.cc	2006-08-15 10:10:51.000000000 +0000
@@ -214,7 +214,7 @@
  ******************************************************************************/
 int main(int argc, char **argv)
 {
-	int chk;
+	int chk,log;
 	char pidnum[8];
 	int _debug, c, errflg;
 	const char *configfile=PXECONFIGFILE;
@@ -258,6 +258,13 @@
 		debug.close();
 	}
 
+	// check to see if the daemon is already running
+	chk = open(LOCKFILE, O_WRONLY|O_CREAT|O_EXCL, 0666);
+	if(-1 == chk)
+	{
+		std::cerr << "PXE daemon already running\n";
+		return(-1);
+	}
 
 	// set the UID/GID to a low user
 #ifndef NO_SUID
@@ -269,6 +276,9 @@
 		     << ", continuing with user id " << getuid() << "\n";
 	else
 	{
+	  if(log=open(PXELOGFILE, O_CREAT, 0666)) close(log);
+	  if(chown(PXELOGFILE,pw->pw_uid,pw->pw_gid)<0)
+	    std::cout << "Unable to acquire logfile " << PXELOGFILE << "\n";
 		if((-1 == setgid(pw->pw_gid)) || (-1 == setegid(pw->pw_gid)))
 			std::cout << "Unable to change group id, continuing with group id "
 			     << getgid() << "\n";
@@ -278,14 +288,6 @@
 	}
 #endif
 
-	// check to see if the daemon is already running
-	chk = open(LOCKFILE, O_WRONLY|O_CREAT|O_EXCL, 0644);
-	if(-1 == chk)
-	{
-		std::cerr << "PXE daemon already running\n";
-		return(-1);
-	}
-
 	// if not in debug mode, fork and go
 	if (0 == _debug) {
 		signal(SIGCHLD, SIG_IGN);
